services/src/main/java/org/keycloak/forms/login/freemarker/FreeMarkerLoginFormsProvider.java

```diff
@@ -143,6 +144,7 @@ private Response createResponse(LoginFormsPages page) {
         RealmModel realm = session.getContext().getRealm();
         ClientModel client = session.getContext().getClient();
         UriInfo uriInfo = session.getContext().getUri();
+        String loginHint = uriInfo.getQueryParameters().getFirst(OIDCLoginProtocol.LOGIN_HINT_PARAM);
```

Comment:
This shouldn't be needed. In fact you can't rely on the query param being set at this point due to the way authentication flows work.